stetho/src/main/java/com/facebook/stetho/inspector/elements/ElementInfo.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.elements;
+
+import com.facebook.stetho.common.ListUtil;
+import com.facebook.stetho.common.Util;
+
+import java.util.List;
+
+public final class ElementInfo {
+  public final Object element;
+  public final Object parentElement;
+  public final List<Object> children;
+
+  public ElementInfo(
+      Object element,
+      Object parentElement,
+      List<Object> children) {
+    this.element = Util.throwIfNull(element);
+    this.parentElement = parentElement;
+    this.children = ListUtil.copyToImmutableList(children);
```

Comment:
The grating thing for me here is that I just can't tell if this is a copy or not.  Or if the copy is conditional (sometimes it's a copy but usually it's not?)  I find it considerably more clear to just enforce these contracts through design and documentation when they are _INTERNAL_ to your own module.

I'm fine to let it land as-is, but I can tell you from experience this is not a pattern that mobile wants.